source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,904 @@
+
+Terms
+-----
+
+min(a, b)
+  Shorthand for "the minimum of a and b" where ``a`` and ``b`` are numeric
+  values. For any cases where 0 means "infinite" (e.g. `timeoutMS`_),
+  ``min(0, other)`` MUST evaluate to ``other``.
+
+MongoClient Configuration
+-------------------------
+
+This specification introduces a new configuration option and deprecates some
+existing options.
+
+timeoutMS
+~~~~~~~~~
+
+This 64-bit integer option specifies the per-operation timeout value in
+milliseconds. The default value is unset. Both unset and an explicit value of
+0 mean infinite, though some client-side timeouts like
+``serverSelectionTimeoutMS`` will still apply. Drivers MUST error if a
+negative value is specified. This value MUST be configurable at the level of
+a MongoClient, MongoDatabase, MongoCollection, or of a single operation.
+However, if the option is specified at any level, it cannot be later changed
+to unset. At each level, the value MUST be inherited from the previous level
+if it is not explicitly specified.
+
+See `timeoutMS cannot be changed to unset once it’s specified`_.
+
+Backwards Breaking Considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This specification deprecates many existing timeout options and introduces a
+new exception type that is used to communicate timeout expiration. If drivers
+need to make backwards-breaking changes to support ``timeoutMS``, the
+backwards breaking behavior MUST be gated behind the presence of the
+``timeoutMS`` option. If the ``timeoutMS`` option is not set, drivers MUST
+continue to honor existing timeouts such as ``socketTimeoutMS``. Backwards
+breaking changes include any changes to exception types thrown by stable API
+methods or changes to timeout behavior. Drivers MUST document these changes.
+
+In a subsequent major release, drivers SHOULD drop support for legacy timeout
+behavior and only continue to support the timeout options that are not
+deprecated by this specification. Once legacy options are removed, drivers
+MUST make the backwards-breaking behavioral changes described in this
+specification regardless of whether or not ``timeoutMS`` is set by the
+application.
+
+See the `Errors`_ section for explanations of the backwards-breaking changes
+to error reporting.
+
+Deprecations
+~~~~~~~~~~~~
+
+The following configuration timeout options MUST be deprecated in favor of
+``timeoutMS``:
+
+- ``socketTimeoutMS``
+
+- ``waitQueueTimeoutMS``
+
+- ``wTimeoutMS``
+
+The following options for CRUD methods MUST be deprecated in favor of
+``timeoutMS``:
+
+- ``maxTimeMS``
+
+- ``maxCommitTimeMS``
+
+Drivers MUST throw an error during MongoClient configuration if ``timeoutMS``
+is used in conjunction with one of the options in the deprecated list. This
+validation MUST NOT be done during MongoDatabase or MongoCollection
+configuration.
+
+See `Option validation does not occur at the database or collection levels`__
+and `Option validation does not occur at the database or collection levels`__.
```

Comment:
These two links are the same.

Reply:
Yeah, this was supposed to be two separate links. Fixed here and in below in the `Validation and Overrides` section.